src/Elastic.Apm/Model/Span.cs

```diff
@@ -734,7 +734,7 @@ private void DeduceServiceTarget()
 			if (!IsExitSpan)
 				return;
 
-			if (!_context.IsValueCreated)
+			if (!_context.IsValueCreated && !IsExitSpan)
```

Comment:
- What here actually happens is that we store `Context.Service.Target` (and resource) on the top level in `DroppedSpanStatCache` for dropped spans and then return - with this we avoid initialization of the huge `Context` object, which would happen later in the method. At least that's how I interpret this :)  So... shouldn't be an `&& _isDropped` in this check? Because if the span is not dropped, we don't want to set `DroppedSpanStatCache`.
- I feel a one liner comment would help here a lot. It took me a while to understand this :) Maybe let's add something like this before the line (just an idea, feel free to change the text): "In order to avoid the creation of `Context`, set `target` and `resource` on the top level `DroppedSpanStatCache` and return".

Reply:
* You mean, this check for `_isDropped` should have been here in the first place? But yes, I'll add it locally and run the tests again.
* Yes, an explanatory comment is definitely a good idea here.

Reply:
> You mean, this check for _isDropped should have been here in the first place? 

Yes. So the original check wasn't very correct - I guess that code was from me.

Reply:
> > You mean, this check for _isDropped should have been here in the first place?
> 
> Yes. So the original check wasn't very correct - I guess that code was from me.

No blaming ;-) - this was just for my understanding.